programs/disks/CommandCopy.h

```diff
@@ -0,0 +1,61 @@
+#pragma once
+
+#include "ICommand.h"
+namespace DB
+{
+
+namespace ErrorCodes
+{
+    extern const int BAD_ARGUMENTS;
+}
+
+class CommandCopy : public ICommand
+{
+public:
+    CommandCopy() = default;
+    CommandCopy(std::optional<ProgramOptionsDescription> & command_option_description_input)
+    {
+        command_name = "copy";
+        command_option_description.emplace(createOptionsDescription("Help Message for copy", getTerminalWidth()));
+        description = "Recursively copy data containing at `from_path` to `to_path`";
+        usage = "Usage: copy [OPTION]... <FROM_PATH> <TO_PATH>";
+        command_option_description->add(command_option_description_input.value());
+        command_option_description->add_options()
+            ("help,h", "print help message for copy");
+    }
+
+    void execute(
+        const std::vector<String> & command_arguments,
+        const ContextMutablePtr & global_context,
+        const Poco::Util::AbstractConfiguration & config) override
+    {
+        if (command_arguments.size() != 3)
+        {
+            printHelpMessage();
+            throw DB::Exception("Bad Arguments", DB::ErrorCodes::BAD_ARGUMENTS);
+        }
+
+        String disk_name_from = config.getString("diskFrom", config.getString("disk", "default"));
+        String disk_name_to = config.getString("diskTo", config.getString("disk", "default"));
+
+        String path_from = command_arguments[1];
+        String path_to = command_arguments[2];
+        if (path_from[0] == '/')
+            path_from.erase(0, 1);
+        if (path_to[0] == '/')
+            path_to.erase(0, 1);
+
+        DiskPtr disk_from = global_context->getDisk(disk_name_from);
+        DiskPtr disk_to = global_context->getDisk(disk_name_to);
+
+        String full_path_from = (disk_name_from == "default"
+                                     ? ((fs::path(disk_from->getPath()) / "data/default") / path_from)
+                                     : (fs::path(disk_from->getPath()) / path_from));
+        String full_path_to = (disk_name_to == "default"
+                                   ? ((fs::path(disk_to->getPath()) / "data/default") / path_to)
+                                   : (fs::path(disk_to->getPath()) / path_to));
```

Comment:
why need to add "data/default" ? `fs::path(disk->getPath()) / path` should suffice for any case